Minter_v1.0.1.sol

```diff
@@ -517,30 +550,22 @@ contract SharedDeposit is Pausable, ReentrancyGuard {
         BETHToken.mint(msg.sender, valMinusAdmin);
     }
 
-    function withdraw(uint256 amount) public nonReentrant whenNotPaused {
-        uint256 valBeforeAdmin =
-            amount.mul(1e18).div(
+    function withdraw(uint256 amount) external nonReentrant whenNotPaused {
+        uint256 valBeforeAdmin;
+        if (disableWithdrawRefund) {
+            valBeforeAdmin = amount;
+        } else {
+            valBeforeAdmin = amount.mul(1e18).div(
                 uint256(1).mul(1e18).sub(
-                    adminFee.mul(1e18).div(costPerValidator)
+                   adminFee.mul(1e18).div(costPerValidator)
                 )
             );
-        if (disableWithdrawRefund == true) {
-            valBeforeAdmin = amount;
         }
         uint256 newShareTotal = curValidatorShares.sub(amount);
-
-        require(
```

Comment:
As the withdraw function burns vETH2 to return ETH, it is impossible for it to increase the `newShareTotal`. Therefore this require is basically a no-op.